services/observer.js

```diff
@@ -65,4 +65,12 @@ require('core/transport')
             this.send(events.CHAINDATA_SERVER_CREATED, port, msg.sender);
         }
     })
+
+    .on(events.PING, function areYouLookingForMe(data, msg) {
+        const addr = wallet.address.toString('hex');
+
+        if (data === addr) {
```

Comment:
Think of different naming. Since it's only address it should be `addrReceived` or something like that. `data` associates with something more generic or object, at least.